finality-aleph/src/network/clique/protocols/v0/mod.rs

```diff
@@ -98,6 +102,17 @@ pub async fn incoming<SK: SecretKey, D: Data, S: Splittable>(
         "Incoming handshake with {} finished successfully.", public_key
     );
 
+    let authorized = handle_authorization::<SK>(authorization_requests_sender, public_key.clone())
+        .await
+        .map_err(|_| ProtocolError::NotAuthorized)?;
+    if !authorized {
```

Comment:
There is some double-logic happening here, the error above is called `NotAuthorized`, but is actually about authorization failing for other reasons, and then here the actual lack of authorization happens.

To be more in line with the rest of this function I would suggest having `handle_authorization` return `ProtocolError`, in practice either `NoParentConnection` or `NotAuthorized`. Then don't log it here, but just `?` it. The function wouldn't have to return a `bool` at all.

I think it would also be simpler if you didn't send a `bool` through the oneshot, just a unit, and use dropping to signal lack of interest to have less states to worry about, like the surrounding code does. However, if you think sending a `bool` provides useful information, then do it, but return `NoParentConnection` on a closed channel and `NotAuthorized` on a `false`.

Reply:
Changed.